lib/runHook.js

```diff
@@ -5,91 +5,55 @@ var path = require('path');
-        return filename;
-    }
-
-    // When doing livereload, at least one platform has to successfully launch, otherwise we just abort ...
-    var anyPlatformSucceeded = false;
+    });
 
-    return Q()
-        .then(function() {
+    // If none of the supplied platforms are supported, stop and return an error
+    if (options.platforms === undefined || options.platforms.length == 0) {
+        return Q.reject(pluginName + ': None of the platforms supplied are currently supported.');
+    }
 
-            // Start LiveReload on all specified platforms serially, so that we don't collide over free ports
-            // ... Also, if any platform fails, print a warning and carry on with the other ones.
-            return promiseUtils.Q_chainmap_graceful(options.platforms, function(platform) {
+    var platform_wwws = options.platforms.map(function(plat) {
+        return path.join(projectRoot, multiPlatforms.getPlatformWWWFolder(plat)); 
+    });
 
-                var platform_www = path.join(projectRoot, WWW_FOLDER[platform]);
-                var configXml = path.join(projectRoot, CONFIG_LOCATION[platform], 'config.xml');
 
-                var LiveReload = require('./livereload');
-                return new LiveReload(projectRoot, platform_www, options, context).StartServer().then(function(server_url) {
-                    var platformIndexUrl = url.resolve(server_url, path.join(WWW_FOLDER[platform], 'index.html'));
-                    changeConfigXml(platformIndexUrl, configXml);
-                    return Q();
-                }).then(function() {
-                    // Remove CSP to allow calls to BrowserSync HTTP server
-                    var platformIndexLocal = path.join(platform_www, 'index.html');
-                    var cspRemover = new CSPRemover(platformIndexLocal);
-                    return cspRemover.Remove();
-                }).then(function() {
-                    anyPlatformSucceeded = true;
-                });
-            }, function(err) {
-                events.emit('warn', err);
-            });
-        })
-        .then(function() {
-            // When doing livereload, at least one platform has to successfully launch, otherwise we just abort ...
-            if (!anyPlatformSucceeded) {
-                var msg = 'cordova-plugin-livereload: Livereload failed on all platforms, thus the "run" process is being aborted ... ';
-                events.emit('warn', msg);
-                return Q.reject(msg);
-            }
-            return Q();
-        }, function(err) {
-            events.emit('warn', err);
+    var LiveReload = require('./livereload');
+    return new LiveReload(projectRoot, platform_wwws, options, context.cordova.raw.prepare).StartServer().then(function(server_url) {
+        return promiseUtils.Q_chainmap(options.platforms, function(plat) {
+            var platformIndexUrl = url.resolve(server_url, path.join(multiPlatforms.getPlatformWWWFolder(plat), 'index.html'));
```

Comment:
what if the start page for an app is other than index.html? We should probably read the name of the current start page and then append it here - instead of hardcoding index.html.